rancilio-pid/rancilio-pid.ino

```diff
@@ -1366,14 +1547,31 @@ void loop() {
       if (pidMode == 1) bPID.SetMode(AUTOMATIC);
       bPID.SetTunings(aggoKp, aggoKi, aggoKd);
 
+    /* state 6: Steam mode active*/
+    } else if (activeState == 6) {
+      if (aggSteamTn != 0) {
+        aggSteamKi = aggSteamKp / aggSteamTn ;
+      } else {
+        aggSteamKi = 0;
+      }
+      aggSteamKd = aggSteamTv * aggSteamKp ;
```

Comment:
Fehlt hier nicht ein: bPID.SetTunings(aggSteamKp, aggSteamKi, aggSteamKd) ?

2. Frage: Ist esüberhaupt notwendig eigene Steam-Parameter zu haben? Je weniger Settings konfigurierbar sind, desto besser.

Reply:
also Kp muss bei mir geändert werden sonst kommt die Maschine nicht auf den setpointSteam in einigermaßen vertretbarer Zeit. Bei mir ist das z.B.: auch Kp = 1000.

Reply:
1. du meinst Kd und nicht Kp, oder?
2. Unabhängig davon berechnest du zwar die PID Werte, aber setzt sie nicht was man mit dem bPID.SetTunings() tun würde. 
Überseh ich was?

Reply:
hoffe dass ich es nicht zu sehr vereinfache und es dann nicht mehr funktioniert :-) Bitte nicht schlachten ;-)

Reply:
> 1. du meinst Kd und nicht Kp, oder?
> 2. Unabhängig davon berechnest du zwar die PID Werte, aber setzt sie nicht in dem State 6 (dh wenn du aufheizen willst) was man mit dem bPID.SetTunings() tun würde.
>    Überseh ich was?
>    Achso.. du willst diese PID Parameter nicht zum aufheizen sondern um abkühlen verwenden..

nee ich meinte Kp wenn ich auf dem Dampfpunkt hoch heize, kd wirkt ja gegenteilig. Damit die Maschine möglichst lange mit viel Leistung heizt habe ich kp auf 1000 gesetzt.

Ich möchte die schon zum aufheizen, werden aber in der Funktion generateSteam() gesetzt und in die loop nur überprüft. Beim abkühlen setzte ich Kd auf null, weil da bei mir kleine Werte zum starken heizen beim abkühlen geführt haben. Und um den User vor Fehleingaben zu schützen. Bei mir hat sich nach dem Steamen die Maschine mit geöffnetem Dampfhahn bei 106°C eingependelt...